tests/KubernetesClient.Tests/AuthTests.cs

```diff
@@ -162,15 +162,10 @@ public void BasicAuth()
         }
 
 #if NETCOREAPP2_1 // The functionality under test, here, is dependent on managed HTTP / WebSocket in .NET Core 2.1 or newer.
-        [Fact]
+        // this test doesn't work on OSX and is inconsistent on windows
```

Comment:
Is there a way to fix this test so that it works on Windows? What is the underlying problem?

Reply:
Here's a failed build run where this "random" failure shows up
https://github.com/kubernetes-client/csharp/runs/613358063?check_suite_focus=true

```
 total of 1 test files matched the specified pattern.
[xUnit.net 00:00:09.58]     k8s.Tests.WatchTests.DirectWatchEventsWithTimeout [SKIP]
[xUnit.net 00:00:09.82]     k8s.Tests.WatchTests.WatcherIntegrationTest [SKIP]
  ! k8s.Tests.WatchTests.DirectWatchEventsWithTimeout [1ms]
  ! k8s.Tests.WatchTests.WatcherIntegrationTest [1ms]
[xUnit.net 00:02:11.62]     k8s.Tests.AuthTests.Cert [FAIL]
  X k8s.Tests.AuthTests.Cert [270ms]
  Error Message:
   System.AggregateException : One or more errors occurred. (The SSL connection could not be established, see inner exception.)
---- System.Net.Http.HttpRequestException : The SSL connection could not be established, see inner exception.
-------- System.ComponentModel.Win32Exception : The credentials supplied to the package were not recognized
  Stack Trace:
     at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at k8s.Tests.AuthTests.ExecuteListPods(IKubernetes client) in D:\a\csharp\csharp\tests\KubernetesClient.Tests\AuthTests.cs:line 37
   at k8s.Tests.AuthTests.Cert() in D:\a\csharp\csharp\tests\KubernetesClient.Tests\AuthTests.cs:line 223
----- Inner Stack Trace -----
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at k8s.WatcherDelegatingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in 
   at System.Net.Security.SslStreamPal.AcquireCredentialsHandle(X509Certificate certificate, SslProtocols protocols, EncryptionPolicy policy, Boolean isServer)
   at System.Net.Security.SecureChannel.AcquireClientCredentials(Byte[]& thumbPrint)
   at System.Net.Security.SecureChannel.GenerateToken(Byte[] input, Int32 offset, Int32 count, Byte[]& output)
   at System.Net.Security.SecureChannel.NextMessage(Byte[] incoming, Int32 offset, Int32 count)
   at System.Net.Security.SslState.StartSendBlob(Byte[] incoming, Int32 count, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.PartialFrameCallback(AsyncProtocolRequest asyncRequest)
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Security.SslState.ThrowIfExceptional()
   at System.Net.Security.SslState.InternalEndProcessAuthentication(LazyAsyncResult lazyResult)
   at System.Net.Security.SslState.EndProcessAuthentication(IAsyncResult result)
   at System.Net.Security.SslStream.EndAuthenticateAsClient(IAsyncResult asyncResult)
   at System.Net.Security.SslStream.<>c.<AuthenticateAsClientAsync>b__47_1(IAsyncResult iar)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)
  Standard Output Messages:
 [Information] HttpsConnectionAdapter: Failed to authenticate HTTPS connection.
```

I cannot reproduce this on my local machine under any framework, this seems to OCCASIONALLY happen on the windows worker. It seems like an internal SSPI exception that is rejecting cert. On Linux, it would be using OpenSSL.

Reply:
we do have mac ci test
`.NET Core / Dotnet build (macOS-latest) (pull_request)`
any changes we need for mac ci?